sematic/plugins/metrics_storage/sql/sql_metrics_storage.py

```diff
@@ -0,0 +1,241 @@
+            session.commit()
+
+            session.add_all(metric_values)
+            session.commit()
+
+    def get_metrics(self, labels: MetricsLabels) -> Tuple[str, ...]:
+        with db().get_session() as session:
+            metric_names = (
+                session.query(MetricLabel.metric_name)
+                .distinct(MetricLabel.metric_name)
+                .filter(*_make_predicates_from_labels(labels))
+                .all()
+            )
+
+        return tuple(row[0] for row in metric_names)
+
+    def get_aggregated_metrics(
+        self, filter: MetricsFilter, group_by: Sequence[GroupBy]
+    ) -> MetricSeries:
+        # Early check as later queries fail when there are no rows.
+        try:
+            with db().get_session() as session:
+                session.query(MetricLabel.metric_id).filter(
+                    MetricLabel.metric_name == filter.name
+                ).limit(1).one()
+        except NoResultFound:
+            raise NoMetricError(filter.name, self.get_path())
+
+        predicates = [
+            MetricLabel.metric_id == MetricValue.metric_id,
+            MetricLabel.metric_name == filter.name,
+            MetricValue.metric_time > filter.from_time,
+            MetricValue.metric_time <= filter.to_time,
+        ]
+
+        predicates += _make_predicates_from_labels(filter.labels)
+
+        select_fields = [
+            MetricLabel.metric_name,
+            MetricLabel.metric_type,
+            func.sum(MetricValue.value),
+            func.count(MetricValue.value),
+        ]
+
+        n_basic_fields = len(select_fields)
+
+        extra_field_names = []
+
+        group_by_clauses = [
+            MetricLabel.metric_name,
+            MetricLabel.metric_type,
+        ]
+
+        for gb in group_by:
+            extra_field_names.append(gb.value)
+
+            if gb is GroupBy.date:
+                field_ = func.date(MetricValue.metric_time)
+            elif gb is GroupBy.timestamp:
+                field_ = func.extract("epoch", MetricValue.metric_time)
+            else:
+                field_ = MetricLabel.metric_labels[gb.value].astext
+
+            select_fields.append(field_)
+            group_by_clauses.append(field_)
+
+        with db().get_session() as session:
+            query = session.query(*select_fields).filter(*predicates)
+
+            if len(group_by_clauses) > 0:
+                query = query.group_by(*group_by_clauses)
+
+            records = query.all()
+
+        output = MetricSeries(
+            metric_name=filter.name, group_by_labels=extra_field_names
+        )
+
+        for record in records:
+            _, metric_type, metric_sum, metric_count = record[:n_basic_fields]
+            output.metric_type = MetricType(metric_type).name
+            metric_value: float = metric_sum
+            if metric_type == MetricType.GAUGE.value:
+                metric_value = float(metric_sum) / (metric_count or 1)
```

Comment:
This supports histograms. As long as values are floats. Every histogram value can be mapped to a float. The SQL query would have to be different though.

Regarding this handling, this is an optimization. I could use `func.avg` instead of `func.sum` on line 131. But I would have to query the one metric value first to know whether the metric is a gauge or count. I could also compute both sum and avg and drop the one that is irrelevant.